src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -234,8 +251,14 @@ Status CoreWorkerPlasmaStoreProvider::Wait(
       should_break = remaining_timeout <= 0;
     }
 
-    RAY_RETURN_NOT_OK(raylet_client_->Wait(id_vector, num_objects, call_timeout, false,
-                                           task_id, &result_pair));
+    // This is a separate IPC from the Wait in direct call mode.
+    if (ctx.CurrentTaskIsDirectCall() && ctx.ShouldReleaseResourcesOnBlockingCalls()) {
+      RAY_RETURN_NOT_OK(raylet_client_->NotifyDirectCallTaskBlocked());
+    }
+    RAY_RETURN_NOT_OK(
+        raylet_client_->Wait(id_vector, num_objects, call_timeout, false,
```

Comment:
Just wondering is it required for `NotifyDirectCallTaskBlocked` to be called before `Wait`?  

Would it work if we reverse the order of the two operations? I'm not suggest this change, but just curious.

Reply:
Wait is a blocking call, so I think it would basically mean the block call would become a no-op if done afterwards.

Reply:
I was just thinking of the out of order issue of grpc - it's fine for now since we're still using raw tcp connection to communicate with raylet, but if later we want to migrate to grpc then this could be an issue. I saw @stephanie-wang had a similar comment.